### PR TITLE
Remove kv.snapshot_recovery.max_rate setting

### PR DESCRIPTION
**Pull request by rmloveland**
Fixes DOC-9566

### CODE REVIEW

**Comment by rmloveland** [comment]
@andrewbaptist thanks for adding the release note on https://github.com/cockroachdb/cockroach/pull/102596 - for some reason our automation didn't create a docs issue for this, but I just found it due to a current Slack convo about these settings, hence this PR

**Review by andrewbaptist — APPROVED** [review]
:lgtm:

File: src/current/v23.2/cluster-setup-troubleshooting.md [hunk 1]
```diff
@@ -170,28 +170,23 @@ W180817 17:01:56.510430 914 vendor/google.golang.org/grpc/clientconn.go:1293 grp
 
 ###### Excessive snapshot rebalance and recovery rates
 
-The `kv.snapshot_rebalance.max_rate` and `kv.snapshot_recovery.max_rate` [cluster settings]({% link {{ page.version.version }}/cluster-settings.md %}) set the rate limits at which [snapshots]({% link {{ page.version.version }}/architecture/replication-layer.md %}#snapshots) are sent to nodes. These settings can be temporarily increased to expedite replication during an outage or when scaling a cluster up or down.
+The `kv.snapshot_rebalance.max_rate` [cluster setting]({% link {{ page.version.version }}/cluster-settings.md %}#setting-kv-snapshot-rebalance-max-rate) sets the rate limits at which [snapshots]({% link {{ page.version.version }}/architecture/replication-layer.md %}#snapshots) are sent to nodes. This setting can be temporarily increased to expedite replication during an outage or when scaling a cluster up or down.
```
Comment (taroface) [inline src/current/v23.2/cluster-setup-troubleshooting.md:173]:
```suggestion
The `kv.snapshot_rebalance.max_rate` [cluster setting]({% link {{ page.version.version }}/cluster-settings.md %}#setting-kv-snapshot-rebalance-max-rate) sets the rate limit at which [snapshots]({% link {{ page.version.version }}/architecture/replication-layer.md %}#snapshots) are sent to nodes. This setting can be temporarily increased to expedite replication during an outage or when scaling a cluster up or down.
```

**Review by taroface — APPROVED** [review]
LGTM with one tiny suggestion

File: src/current/v23.2/cluster-setup-troubleshooting.md (hunk 1 above)
Comment (rmloveland) [reply src/current/v23.2/cluster-setup-troubleshooting.md]: Fixed!

**Pull request merged**